ColumbusStateWorkforceInnovation/java-4-degrees-vue-app
# Update to node 14.17.0

## Context
The version of Node currently installed on the VMs that students are using is out-dated. This PR pins the version of Node to the (current) latest Long-Term Support version of 14.17.0.

## Changes
- [X] Pinned the version of node in `.nvmrc`
- [X] `~>` pinned the Vue development dependencies to `4.5.13`.
- [X] Ran `npm audit fix` to correct several hundred security vulnerabilities. We still have more than 100 vulnerabilities to address which are outside the scope of this PR.
- [X] Updated the README to include instructions for installing and using nvm.

## Validation
1. Run `npm install && npm run serve` and validate that the dependencies install correctly and that the frontend application runs as expected.
2. Insure the postgres-db docker image is running.
3. Insert the expected for records the Java IV, Lab 5 solution (see the lab description).
4. Run the Java IV, Lab 5 solution (be sure to activate the postgres profile).
5. Navigate to http://localhost:8080/#/menu and validate that the menu and menu items are populated as expected.
